sigma/backends/elasticsearch/elasticsearch_eql.py

```diff
@@ -0,0 +1,445 @@
+import re
+import json
+from typing import ClassVar, Dict, List, Optional, Pattern, Tuple, Union, Any
+
+from sigma.conversion.state import ConversionState
+from sigma.rule import SigmaRule
+from sigma.conversion.base import TextQueryBackend
+from sigma.conversion.deferred import DeferredQueryExpression
+from sigma.conditions import (
+    ConditionItem,
+    ConditionAND,
+    ConditionOR,
+    ConditionNOT,
+    ConditionFieldEqualsValueExpression,
+)
+from sigma.types import SigmaCompareExpression, SigmaNull, SpecialChars, SigmaNumber
+import ipaddress
+import sigma
+
+
+class EqlBackend(TextQueryBackend):
+    """
+    Elasticsearch event querly language backend. Generates query strings described here in the
+    Elasticsearch documentation:
+
+    https://www.elastic.co/guide/en/elasticsearch/reference/current/eql.html
+    """
+
+    # A descriptive name of the backend
+    name: ClassVar[str] = "Elasticsearch EQL"
+    # Output formats provided by the backend as name -> description mapping.
+    # The name should match to finalize_output_<name>.
+    formats: ClassVar[Dict[str, str]] = {
+        "default": "Plain Elasticsearch EQL queries.",
+        "eqlapi": "Plain EQL queries ready for '/_eql/search' API endpoint.",
+        "siem_rule": "Elasticsearch EQL queries as SIEM Rule.",
+        "siem_rule_ndjson": "Elasticsearch EQL Query as SIEM Rules in NDJSON Format.",
```

Comment:
Perhaps something that could be replaced with post-processing feature from pySigma >= 0.10.

Reply:
I'll do that as a seperate issue - https://github.com/SigmaHQ/pySigma-backend-elasticsearch/issues/33